crates/env/src/types.rs

```diff
@@ -94,6 +94,15 @@ impl FromLittleEndian for u128 {
     }
 }
 
+/// The trait shows that the type plans to be [`Environment::AccountId`].
+///
+/// If someone wants to use another type than [`AccountId`] in the [`Environment::AccountId`],
+/// he should implement this trait for the corresponding type.
```

Comment:
That doesn't really answer my question though. Why would a user want to implement this for a type other an `AccountId`? 

If we don't want this, but don't want to introduce a breaking change, then we shouldn't be highlighting the fact that they _can_ implement this trait on other types

Reply:
The user wants to implement it if he uses another type for the `Environment::AccountId` than `ink::types::AccountId`.

We want this, because it will give the `From<AccountId>` implementation to the user. If it is not a problem to add constrain for `Environment::AccountId`, I prefer to add it.

Reply:
Okay I see what you mean now. I'm gonna update the comment to make this more clear